pkg/services/policer/queue.go

```diff
@@ -19,9 +21,16 @@ func (q *jobQueue) Select(limit int) ([]*object.Address, error) {
 		return nil, err
 	}
 
-	if len(res) < limit {
-		return res, nil
+	num := len(res)
+
+	if num > limit {
+		num = limit
+		res = res[:limit]
 	}
 
-	return res[:limit], nil
+	rand.Shuffle(num, func(i, j int) {
```

Comment:
Shuffling before cutting provides more entropy, isn't it? 

If node stores 1 000 objects and it is able to replicate 100 of them, then it is going to replicate the same 100 objects but in different order. Not sure this is expected solution.